var/spack/repos/builtin/packages/nvhpc/package.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+#
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+
+from spack import *
+from spack.util.prefix import Prefix
+import os
+import platform
+
+# FIXME Remove hack for polymorphic versions
+# This package uses a ugly hack to be able to dispatch, given the same
+# version, to different binary packages based on the platform that is
+# running spack. See #13827 for context.
+# If you need to add a new version, please be aware that:
+#  - versions in the following dict are automatically added to the package
+#  - version tuple must be in the form (checksum, url)
+#  - checksum must be sha256
+#  - package key must be in the form '{os}-{arch}' where 'os' is in the
+#    format returned by platform.system() and 'arch' by platform.machine()
+_versions = {
+    '20.9': {
+        'Linux-aarch64': ('3bfb3d17f5ee99998bcc30d738e818d3b94b828e2d8da7db48bf152a01e22023', 'https://developer.download.nvidia.com/hpc-sdk/20.9/nvhpc_2020_209_Linux_aarch64_cuda_11.0.tar.gz'),
+        'Linux-ppc64le': ('b2966d4047e1dfd981ce63b333ab9c0acbdc2a6a505fa217456ac9fa3b8e7474', 'https://developer.download.nvidia.com/hpc-sdk/20.9/nvhpc_2020_209_Linux_ppc64le_cuda_multi.tar.gz'),
+        'Linux-x86_64': ('fe665ab611b03846a90bd70ca4e08c1e59ab527364b971ed0304e0ae73c778d8', 'https://developer.download.nvidia.com/hpc-sdk/20.9/nvhpc_2020_209_Linux_x86_64_cuda_multi.tar.gz')},
+    '20.7': {
+        'Linux-aarch64': ('5b83ca1919199ac0aa609309b31c345c5a6453dd3131fddeef9e3ee9059a0e9b', 'https://developer.download.nvidia.com/hpc-sdk/20.7/nvhpc_2020_207_Linux_aarch64_cuda_11.0.tar.gz'),
+        'Linux-ppc64le': ('800ead240bdf61611910b2f6df24ee1d7359377ff3767c923738dd81fcea9312', 'https://developer.download.nvidia.com/hpc-sdk/20.7/nvhpc_2020_207_Linux_ppc64le_cuda_multi.tar.gz'),
+        'Linux-x86_64': ('a5c5c8726d2210f2310a852c6d6e03c9ef8c75e3643e9c94e24909f5e9c2ea7a', 'https://developer.download.nvidia.com/hpc-sdk/20.7/nvhpc_2020_207_Linux_x86_64_cuda_multi.tar.gz')}
+}
+
+
+class Nvhpc(Package):
+    """Nvidia optimizing multi-core x64 compilers for Linux
+    with support for debugging and profiling of local MPI processes."""
+
+    homepage = "http://developer.nvidia.com/hpc-sdk"
+
+    maintainers = ['samcmill']
+
+    for ver, packages in _versions.items():
+        key = "{0}-{1}".format(platform.system(), platform.machine())
+        pkg = packages.get(key)
+        if pkg:
+            version(ver, sha256=pkg[0], url=pkg[1])
+
+    variant('network',   default=False,
+            description="Perform a network install")
+    variant('single',    default=True,
+            description="Perform a single system install")
```

Comment:
It occurred to me that this could be simplified in one of two ways:

Firstly, since there are two boolean variants which must have opposing values, you could just have one boolean variant (e.g. keep `network` and set it to `False` by default) - in that case the `if/elif` in `install` could be an `if/else`.

If you think there is value in explicitly labelling each state, you can also encode this as a single valued variant. The `dims` variant in `boxlib` is an example of this:

```
    variant('dims', default='3', values=('1', '2', '3'), multi=False,
            description='Number of spatial dimensions')
```

In this case it would look like

```
    variant('install-type', default='single', values=('single, 'network'), multi=False,
            description='Network installs are for installations shared by different operating systems')
```

Reply:
Great idea.  I changed it to use the single valued variant approached.